### PR TITLE
Integrate driver blacklisting into per-device struct

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
As we now have a per-device disabled flag we can integrate driver blacklisting into that.

If a driver is marked as blacklisted we just set the driver->disabled flag in case we have a freshly detected device. Here we take care about an existing old-style use_blacklist_driver conf key to keep dt installations with an activated blacklisted driver running.

- some minor debug output mods
- make the blacklist test a gboolean

### CODE REVIEW

File: src/common/opencl.c [hunk 1]
```diff
@@ -475,10 +467,23 @@ static int dt_opencl_device_init(dt_opencl_t *cl, const int dev, cl_device_id *d
   cl->crc = crc32(cl->crc, (const unsigned char *)infostr, strlen(infostr));
 
   const gboolean newdevice = dt_opencl_read_device_config(dev);
+  const gboolean blacklist = dt_opencl_check_driver_blacklist(deviceversion);
```
Comment (TurboGit) [inline src/common/opencl.c:470]: for clarity I would name this variable `is_blacklisted`.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Just a minor proposal. TIA.

File: src/common/opencl.c (hunk 1 above)
Comment (jenshannoschwalm) [reply src/common/opencl.c]: sure

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**